app/CineHub/app/src/main/java/org/cinehub/RegisterActivity.java

```diff
@@ -0,0 +1,111 @@
+package org.cinehub;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.os.Bundle;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.Toast;
+
+public class RegisterActivity extends AppCompatActivity {
+
+    // Declare the variables
+    private EditText et_username;
+    private EditText et_email;
+    private EditText et_password;
+    private EditText et_confirm_password;
+    private Button btn_register;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_register);
+
+        // Initialize the variables
+        et_username = findViewById(R.id.et_username);
+        et_email = findViewById(R.id.et_email);
+        et_password = findViewById(R.id.et_passwd);
+        et_confirm_password = findViewById(R.id.et_passwd_confirm);
+        btn_register = findViewById(R.id.btn_register);
+
+        btn_register.setOnClickListener(v -> {
+            String username = et_username.getText().toString().trim();
+            String email = et_email.getText().toString().trim();
+            String password = et_password.getText().toString().trim();
+            String confirmPassword = et_confirm_password.getText().toString().trim();
+
+            boolean isValid = true;
+
+            if (!isUsernameValid(username)) {
+                Toast.makeText(this, "The username is invalid", Toast.LENGTH_SHORT).show();
+                isValid = false;
+            }
+
+            if (!isPasswordSecure(password)) {
+                Toast.makeText(this, "The password is not secure", Toast.LENGTH_SHORT).show();
+                isValid = false;
+            }
+
+            if (!arePasswordsEqual(password, confirmPassword)) {
+                Toast.makeText(this, "The passwords are not equal", Toast.LENGTH_SHORT).show();
+                isValid = false;
+            }
+
+            if (!isEmailValid(email)) {
+                Toast.makeText(this, "The email is invalid", Toast.LENGTH_SHORT).show();
+                isValid = false;
+            }
+
+            if (isValid) {
+                Toast.makeText(this, "The registration was successful", Toast.LENGTH_SHORT).show();
+            }
+        });
+
+    }
+
```

Comment:
Great idea to verify the input using Regex. Could you please refactor those variables into class constants? Maybe later they will be used somewhere else.

I think they will be moved to the model classes once those are implemented.

Reply:
Added! Thanks 👯